cli/tsc/99_main_compiler.js

```diff
@@ -1047,12 +1032,32 @@ delete Object.prototype.__proto__;
     if (logDebug) {
       debug(`serverRequest()`, id, method, args);
     }
-    lastRequestId = id;
-    // reset all memoized source files names
-    scriptFileNamesCache = undefined;
-    // evict all memoized source file versions
-    scriptVersionCache.clear();
     switch (method) {
+      case "$projectChanged": {
+        /** @type {string[]} */
+        const changedScripts = args[0];
+        /** @type {string} */
+        const newProjectVersion = args[1];
+        /** @type {boolean} */
+        const tsConfigChanged = args[2];
+
+        if (tsConfigChanged) {
+          tsConfigCache = null;
+        }
+
+        if (projectVersionCache !== newProjectVersion) {
+          // TODO(nathanwhit): this could be more granular
+          scriptFileNamesCache = undefined;
+        }
+
+        projectVersionCache = newProjectVersion;
+
+        for (const script of changedScripts) {
+          scriptVersionCache.delete(script);
```

Comment:
How is this value hydrated again? Do we have to call `op_script_names` just to get one value? I see this op taking 30-35ms consistently on mid-sized repo (~3000 files in memory).

Reply:
Oooops 🤦 this is unrelated, it's `scriptVersionCache` not `scriptFileNamesCache`